### PR TITLE
[JENKINS-68323] Missing "x" to clear the search bar of the plugin manager on Firefox

### PR DESCRIPTION
**Pull request by frankie139506**
See [JENKINS-68323](https://issues.jenkins.io/browse/JENKINS-68323).
gif1

### Proposed changelog entries

* Search box can clear by "x" on Firefox.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

File: war/src/main/less/form/search.less [hunk 1]
```diff
@@ -135,3 +135,29 @@
...
+	border:1px solid transparent;
+	background-color: transparent;
+	display: block;
+	vertical-align: middle;
+  outline: 0;
+  cursor: pointer;
+}
+
+.close-icon:after {
+	content: "x";
```
Comment (basil) [inline war/src/main/less/form/search.less:149]: I think [U+1F5D9 CANCELLATION X](https://www.fileformat.info/info/unicode/char/1f5d9/index.htm) is a better Unicode glyph for this character than [U+0078 LATIN SMALL LETTER X](https://www.fileformat.info/info/unicode/char/0078/index.htm).

```suggestion
	content: "\1F5D9";
```

(+ whatever other CSS changes are necessary to vertically align the new glyph in the middle)

**Review by NotMyFault — COMMENTED** [review]
Am I testing this somehow wrong? On Firefox 100.0, there's no such "x":

**Comment by frankie139506** [comment]
> Am I testing this somehow wrong? On Firefox 100.0, there's no such "x": 



> Am I testing this somehow wrong? On Firefox 100.0, there's no such "x": 

I can run it in my Firefox and my version is 100.0.
image

gif1

I can run it in my Firefox and my version is 100.0.

**Comment by janfaracik** [comment]
Just my view but I don't think that the missing clear button on Firefox is detrimental enough for us to add it and support it ourselves.

**Review by timja — CHANGES_REQUESTED** [review]
> Just my view but I don't think that the missing clear button on Firefox is detrimental enough for us to add it and support it ourselves.

Agreed, Firefox are slowly working on fixing it... 

----

in chrome on MacOS there's a weird square:

image

Firefox MacOS Monterey can't render the unicode character:

image

**Comment by basil** [comment]
> Firefox MacOS Monterey can't render the unicode character:

Ah that's too bad. Perhaps [U+2715 MULTIPLICATION X](https://www.fileformat.info/info/unicode/char/2715/index.htm) would be better supported.

**Comment by basil** [comment]
> Agreed, Firefox are slowly working on fixing it...

Yeah, I do not feel strongly about this either. If we feel the maintenance burden of this is too high, it may be best to close this PR and leave the Jira ticket open to track confirming that this eventually gets taken care of when Firefox supports this feature.

**Comment by NotMyFault** [comment]
I'm not too opposed to it, but I agree with Jan:
> Just my view but I don't think that the missing clear button on Firefox is detrimental enough for us to add it and support it ourselves.

This doesn't feel like something we necessarily have to address in core itself, but can close the Jira issue once FF release a fix for it.

**Comment by basil** [comment]
Since we have reached a clear consensus, let's close this PR. @frankie139506 Thank you for the contribution, but we do not believe the maintenance burden of a custom implementation is worth the benefit.

**Pull request closed without merging**